src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -277,15 +277,28 @@ public void LogAnalyzerCountSummary()
             ResetDiagnosticLogAggregator();
         }
 
-        private void ResetDiagnosticLogAggregator()
+        public static async Task<IEnumerable<DiagnosticData>> GetDiagnosticsAsync(
+            IDiagnosticAnalyzerService service, Document document, IEnumerable<DiagnosticAnalyzer> analyzers, AnalysisKind kind, CancellationToken cancellationToken)
         {
-            DiagnosticLogAggregator = new DiagnosticLogAggregator(Owner);
+            // given service must be DiagnosticAnalyzerService
+            var diagnosticService = (DiagnosticAnalyzerService)service;
+
+            var analyzerDriverOpt = await CompilationManager.CreateAnalyzerDriverAsync(
+                diagnosticService, document.Project, analyzers, includeSuppressedDiagnostics: false, logAggregatorOpt: null, cancellationToken).ConfigureAwait(false);
+
+            var list = new List<DiagnosticData>();
+            foreach (var analyzer in analyzers)
+            {
+                list.AddRange(await Executor.ComputeDiagnosticsAsync(
+                    diagnosticService, analyzerDriverOpt, document, analyzer, kind, spanOpt: null, logAggregatorOpt: null, cancellationToken).ConfigureAwait(false));
```

Comment:
i saw logAggregator being passed around above, but not through here.  Is that an issue?  can you document in the code why it's ok to not pass this in here?